rust/sleeper_core/src/datafusion/filter_aggregation_config/filter.rs

```diff
@@ -0,0 +1,159 @@
+/*
+* Copyright 2022-2025 Crown Copyright
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+use color_eyre::eyre::{self, ensure};
+use datafusion::{
+    error::Result,
+    logical_expr::{Expr, ScalarUDF, col},
+};
+use filter_udfs::ageoff::AgeOff;
+
+use crate::datafusion::filter_aggregation_config::function_call::FunctionCall;
+use crate::datafusion::filter_aggregation_config::function_reader::FunctionReader;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Filter {
+    /// Skip any row where timestamp in named column is older than `max_age` milliseconds.
+    Ageoff { column: String, max_age: i64 },
+}
+
+impl Filter {
+    /// Creates a filtering expression for this filter instance. The returned
+    /// expression can be passed to [`DataFrame::filter`].
+    pub fn create_filter_expr(&self) -> Result<Expr> {
+        match self {
+            Self::Ageoff { column, max_age } => {
+                Ok(ScalarUDF::from(AgeOff::try_from(*max_age)?).call(vec![col(column)]))
+            }
+        }
+    }
+
+    pub fn parse_config(config_string: &str) -> eyre::Result<Vec<Self>> {
+        let mut reader = FunctionReader::new(config_string);
+        let mut filters = vec![];
+        while let Some(call) = reader.read_function_call()? {
```

Comment:
If the FunctionReader struct implemented Iterator, this loop could be re-written in functional style:
```
let filters=reader().iter().map(FunctionCall::try_into).collect::<Vec<_>>()
Ok(filters)
```
Or inline the first line into the second.